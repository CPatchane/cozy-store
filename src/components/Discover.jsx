import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
