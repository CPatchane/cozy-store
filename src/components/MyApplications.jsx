import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
