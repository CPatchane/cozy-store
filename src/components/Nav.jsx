import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
