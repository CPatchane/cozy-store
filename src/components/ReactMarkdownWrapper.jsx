import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
