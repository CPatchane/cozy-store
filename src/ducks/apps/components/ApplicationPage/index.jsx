/* global cozy */
import React, { Component } from 'react'
import { Link } from 'react-router-dom'

import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'
import withBreakpoints from 'cozy-ui/react/helpers/withBreakpoints'
import Button from 'cozy-ui/react/Button'
import FocusTrap from 'focus-trap-react'

import Header from './Header'
import Gallery from './Gallery'
import Details from './Details'

import {
  preventBackgroundScroll,
  unpreventBackgroundScroll
} from 'lib/scrollHelpers.js'

const MOBILE_PLATFORMS = ['ios', 'android']
const isMobilePlatform = name => MOBILE_PLATFORMS.includes(name)

const { BarCenter } = cozy.bar

export class ApplicationPage extends Component {
  constructor(props) {
    super(props)
    this.state = { displayBarIcon: false, activeTrap: true }
  }

  componentDidMount() {
    preventBackgroundScroll()
    const { mainPageRef } = this.props
    if (mainPageRef && mainPageRef.current) {
      mainPageRef.current.addEventListener('scroll', this.handleScroll, {
        passive: true
      })
    }
  }

  componentWillUnmount() {
    unpreventBackgroundScroll()
    const { mainPageRef } = this.props
    if (mainPageRef && mainPageRef.current) {
      mainPageRef.current.removeEventListener('scroll', this.handleScroll)
    }
  }

  mountTrap = () => {
    this.setState({ activeTrap: true })
  }

  unmountTrap = () => {
    this.setState({ activeTrap: false })
  }

  handleScroll = () => {
    if (
      !this.state.displayBarIcon &&
      this.props.mainPageRef.current.scrollTop > 100
    ) {
      this.setState(() => ({ displayBarIcon: true }))
    } else if (
      this.state.displayBarIcon &&
      this.props.mainPageRef.current.scrollTop < 100
    ) {
      this.setState(() => ({ displayBarIcon: false }))
    }
  }

  render() {
    const {
      t,
      parent,
      app,
      isFetching,
      fetchError,
      breakpoints = {},
      pauseFocusTrap
    } = this.props
    const { displayBarIcon } = this.state
    const { isMobile } = breakpoints
    if (isFetching) {
      return (
        <div className="sto-app">
          <Spinner size="xxlarge" loadingType="appsFetching" middle />
        </div>
      )
    }
    if (fetchError) {
      return (
        <p className="u-error">
          {t('app_modal.install.message.version_error', {
            message: fetchError.message
          })}
        </p>
      )
    }
    const { icon, slug, iconToLoad } = app
    const appName = t(`apps.${app.slug}.name`, { _: app.name })
    const namePrefix = t(`apps.${app.slug}.name_prefix`, {
      _: app.name_prefix || ''
    })
    const appShortDesc = t(`apps.${app.slug}.short_description`, {
      _: app.short_description || ''
    })
    const appLongDesc = t(`apps.${app.slug}.long_description`, {
      _: app.long_description || ''
    })
    const appChanges = t(`apps.${app.slug}.changes`, { _: '' })
    const mobileApps =
      app.platforms &&
      !!app.platforms.length &&
      app.platforms.reduce((mobilePlatforms, platformInfos) => {
        // force to be lower case
        platformInfos.type = platformInfos.type.toLowerCase()
        if (isMobilePlatform(platformInfos.type)) {
          mobilePlatforms.push(platformInfos)
        }
        return mobilePlatforms
      }, [])
    return (
      <FocusTrap
        focusTrapOptions={{
          onDeactivate: this.unmountTrap,
          clickOutsideDeactivates: true
        }}
        paused={pauseFocusTrap}
      >
        <div className="sto-modal-page-app">
          {isMobile &&
            icon &&
            !iconToLoad && (
              <BarCenter>
                <div className="sto-app-bar">
                  <img
                    className={`sto-app-bar-icon ${
                      !displayBarIcon ? 'sto-app-bar-icon--hidden' : ''
                    }`}
                    src={icon}
                    alt={`${slug}-icon`}
                  />
                </div>
              </BarCenter>
            )}
          <div className="sto-app">
            <Button
              icon="back"
              tag={Link}
              to={`${parent}`}
              onClick={this.unmountTrap}
              className="sto-app-back"
              label={t('app_page.back')}
              subtle
            />
            <Header
              app={app}
              namePrefix={namePrefix}
              name={appName}
              description={appShortDesc}
              parent={parent}
            />
            {app.screenshots &&
              !!app.screenshots.length && (
                <Gallery slug={slug} images={app.screenshots} />
              )}
            <Details
              app={app}
              description={appLongDesc}
              changes={appChanges}
              mobileApps={mobileApps}
              parent={parent}
            />
          </div>
        </div>
      </FocusTrap>
    )
  }
}

// translate is needed here for the lang props
export default translate()(withBreakpoints()(ApplicationPage))
