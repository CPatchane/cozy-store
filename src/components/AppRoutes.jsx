import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
