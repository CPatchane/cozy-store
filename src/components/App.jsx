
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
