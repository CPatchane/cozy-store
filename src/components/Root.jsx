import React from 'react'
import { Provider } from 'react-redux'
import { I18n } from 'cozy-ui/react/I18n'
import { HashRouter } from 'react-router-dom'

import App from './App'

const Root = ({ context, lang, store }) => {
  return <I18n lang={lang} dictRequire={(lang) => require(`../locales/${lang}`)}>
    <Provider store={store}>
      <HashRouter>
        <App />
      </HashRouter>
    </Provider>
  </I18n>
}

export default Root
