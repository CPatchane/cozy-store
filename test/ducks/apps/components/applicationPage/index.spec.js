'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from '../../../../jestLib/I18n'
import { ApplicationPage } from 'ducks/apps/components/ApplicationPage'

import mockApp from '../../_mockPhotosRegistryVersion'

const appManifest = mockApp.manifest
const mockError = new Error('This is a test error')

const getProps = (installed, related) => {
  return {
    lang: 'en',
    app: Object.assign({}, appManifest, {
      installed,
      icon: 'https://mockcozy.cc/registry/photos/icon',
      related
    }),
    parent: '/myapps'
  }
}

describe('ApplicationPage component', () => {
  it('should be rendered correctly with provided installed app', () => {
    const props = getProps(true, 'https://photos.mockcozy.cc')
    const component = shallow(
      <ApplicationPage t={tMock} {...props} />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with app infos from registry (not installed)', () => {
    const props = getProps(false, null)
    const component = shallow(
      <ApplicationPage t={tMock} {...props} />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a spinner if isFetching', () => {
    const props = getProps(false, null)
    const component = shallow(
      <ApplicationPage t={tMock} isFetching {...props} />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should render correctly an error message if fetchError', () => {
    const props = getProps(false, null)
    const component = shallow(
      <ApplicationPage t={tMock} fetchError={mockError} {...props} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
