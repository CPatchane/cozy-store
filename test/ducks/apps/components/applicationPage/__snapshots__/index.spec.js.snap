// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationPage component should be rendered correctly with app infos from registry (not installed) 1`] = `
<div>
  <div
    className="sto-app"
  >
    <Unknown
      description="Gestionnaire de photos pour Cozy v3"
      editor="Cozy"
      icon="https://mockcozy.cc/registry/photos/icon"
      installed={false}
      installedAppLink={null}
      name="Photos"
      parent="/myapps"
      slug="photos"
    />
    <Unknown
      images={undefined}
      slug="photos"
    />
    <Unknown
      category="cozy"
      changes="### NEW!!!

 Now on the cozy-store :tada"
      description="### Features

Enjoy features to grab your data in you Cozy :tada:"
      developer={
        Object {
          "name": "Cozy",
          "url": "https://cozy.io",
        }
      }
      langs={
        Array [
          "en",
          "fr",
        ]
      }
      mobileApps={
        Array [
          Object {
            "type": "ios",
            "url": "",
          },
          Object {
            "type": "android",
            "url": "https://mock.app",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`ApplicationPage component should be rendered correctly with provided installed app 1`] = `
<div>
  <div
    className="sto-app"
  >
    <Unknown
      description="Gestionnaire de photos pour Cozy v3"
      editor="Cozy"
      icon="https://mockcozy.cc/registry/photos/icon"
      installed={true}
      installedAppLink="https://photos.mockcozy.cc"
      name="Photos"
      parent="/myapps"
      slug="photos"
    />
    <Unknown
      images={undefined}
      slug="photos"
    />
    <Unknown
      category="cozy"
      changes="### NEW!!!

 Now on the cozy-store :tada"
      description="### Features

Enjoy features to grab your data in you Cozy :tada:"
      developer={
        Object {
          "name": "Cozy",
          "url": "https://cozy.io",
        }
      }
      langs={
        Array [
          "en",
          "fr",
        ]
      }
      mobileApps={
        Array [
          Object {
            "type": "ios",
            "url": "",
          },
          Object {
            "type": "android",
            "url": "https://mock.app",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`ApplicationPage component should render correctly a spinner if isFetching 1`] = `
<div
  className="sto-app"
>
  <Unknown
    loadingType="appsFetching"
    middle="true"
    size="xxlarge"
  />
</div>
`;

exports[`ApplicationPage component should render correctly an error message if fetchError 1`] = `
<p
  class="coz-error"
>
  Something went wrong when fetching the application informations. Reason: This is a test error
</p>
`;
